services/sensors/builtin/builtin.go

```diff
@@ -8,22 +8,20 @@ import (
 	"github.com/pkg/errors"
 
 	"go.viam.com/rdk/components/sensor"
-	"go.viam.com/rdk/internal"
 	"go.viam.com/rdk/logging"
 	"go.viam.com/rdk/resource"
 	"go.viam.com/rdk/services/sensors"
 )
 
 func init() {
-	resource.RegisterDefaultService(sensors.API, resource.DefaultServiceModel, resource.Registration[sensors.Service, resource.NoNativeConfig]{
-		Constructor: NewBuiltIn,
-		// NOTE(erd): this ideally would be a matcher on all resources that
-		// contain a Readings proto method. However, these resource types
-		// are heterogeneous and do not actually have this proto method.
-		// We need to either 1. know in advance what kinds of resources are
-		// sensors 2. use a heuristic to detect it or 3. encode it in proto.
-		WeakDependencies: []internal.ResourceMatcher{internal.ComponentDependencyWildcardMatcher},
-	})
+	resource.RegisterDefaultService(
+		sensors.API,
+		resource.DefaultServiceModel,
+		resource.Registration[sensors.Service, resource.NoNativeConfig]{
+			Constructor:      NewBuiltIn,
+			WeakDependencies: []resource.Matcher{resource.InterfaceMatcher{Interface: new(resource.Sensor)}},
```

Comment:
Ooh I like this idea; the `InterfaceMatcher` is cool! Do all sensors the `sensors` service needs to depend on implement `Readings`? What about modular sensors?

Reply:
Good question.  I assumed yes given the comment that existed.  I'll check in on this though

Reply:
Can you run this with a fake sensor and movement sensor and power sensor and verify that the sensors card can `GetAllReadings` and `GetReadings` on the fake sensors?

The interface used to be "component:*:**" so that includes API methods that are not readings. Not entirely sure if this service connects to the collectors either @agreenb

Reply:
It doesn't look like the builtin sensor service even has collectors so I think we're ok on that front.

I'll run the test you asked for though.

Reply:
Went through this with Rand, works as expected.

Reply:
actually, I thought of another case we should check - can you add a modular sensor written in python like the os_status sensor or wifi sensor, and check the rc card page?

Reply:
I tested this - it also works.

Reply:
Thanks for checking!